adapter/redisquota/redisquota.go

```diff
@@ -130,41 +140,105 @@ func (rq *redisQuota) alloc(args adapter.QuotaArgs, bestEffort bool) (adapter.Qu
 		}
 		defer rq.redisPool.put(conn)
 
-		// increase the value of this key by the amount of result
-		conn.pipeAppend("INCRBY", key, result)
-		ret, err := conn.getIntResp()
-		if err != nil {
-			rq.redisError = rq.common.Logger.Errorf("Unable to get integer response from redis: %v", err)
-			return 0, time.Time{}, 0
-		}
+		// Rolling-window algorithm
+		// For rate-limiting use only, and no release for now.
+		if rq.rateLimit == "rolling-window" {
+			// Subsequent commands between MULTI and EXEC will be queued for atomic execution.
+			// The queued commands will not produce any real response until EXEC.
+			conn.pipeAppend("MULTI")
+			resp, _ := conn.pipeResponse()
```

Comment:
Done. However, a lot of error handlings here could not be not covered by unit tests. And that's the main reason for the low coverage now.